tests/tools.vitruv.dsls.demo.familiespersons.tests/src/tools/vitruv/dsls/demo/familiespersons/tests/FamiliesToPersonsTest.xtend

```diff
@@ -89,23 +136,12 @@ class FamiliesPersonsTest extends VitruvApplicationTest {
 	final static Male SON22 = PersonsFactory.eINSTANCE.createMale => [fullName = FIRST_SON_2 + " " + LAST_NAME_2]
 	final static Female DAU22 = PersonsFactory.eINSTANCE.createFemale => [fullName = FIRST_DAU_2 + " " + LAST_NAME_2]
 
-	/**Set the correct set of reactions and routines for this test suite
-	 */
-	override protected getChangePropagationSpecifications() {
-		return #[new FamiliesToPersonsChangePropagationSpecification()]
-	}
 
-	@BeforeEach
-	def disableTransitiveChangePropagation() {
-		virtualModel.changePropagationMode = ChangePropagationMode.SINGLE_STEP
-	}
-	
-	/**Before each test a new {@link FamilyRegister} is created as starting point.
+	/**
+	 * Before each test a new {@link FamilyRegister} is created as starting point.
```

Comment:
We had to remove the `@BeforeEach` and call the method manually every time to avoid race conditions with other `@BeforeEach` calls?
We could also call this method in the `prepare` method otherwise.

Reply:
The problem is that on the one hand no execution order for `@BeforeEach` methods is guaranteed and, even more important, that `@BeforeEach` methods of superclasses are executed before those of subclasses. Thus, extending the test class in which a different `TestView` is provided (such as in our application tests) does not work when a test model is initialized in the superclass before.

In addition to these technical problems, I find it bad practice to always initialize a magic initial state, which is not made explicit in a test case. The `@BeforeEach` should set up an environment but should not provide some model state that may be used or not. The even better fix would be to set up exactly that model that is required in each test case. Some tests even ignore the model that is set up in the `@BeforeEach` and create their own different models, which confuses the functionality even more.